Library/Formula/vault.rb

```diff
@@ -0,0 +1,35 @@
+require "language/go"
+
+class Vault < Formula
+  homepage "https://vaultproject.io/"
+  url "https://github.com/hashicorp/vault.git",
+      :tag => "v0.1.0",
+      :revision => "c8d2dfa67886a80cb4347c25f332a74f91313d23"
+
+  head "https://github.com/hashicorp/vault.git"
+
+  depends_on "go" => :build
+
+  def install
+    gopath = buildpath/"../gopath"
+
+    mkdir_p gopath
+    ENV["GOPATH"] = gopath
+    ENV.prepend_create_path "PATH", gopath/"bin"
+
+    system "go", "get", "github.com/tools/godep"
+    system "go", "get", "github.com/mitchellh/gox"
+
+    mkdir_p gopath/"src/github.com/hashicorp"
+    cp_r buildpath, gopath/"src/github.com/hashicorp/vault"
+
+    cd gopath/"src/github.com/hashicorp/vault" do
+      system "make", "dev"
+      bin.install "bin/vault"
+    end
+  end
+
+  test do
+    system bin/"vault", "--version"
```

Comment:
Can a test be added to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!